source/fe/fe_poly_tensor.cc

```diff
@@ -49,30 +49,29 @@ namespace internal
        * implements an algorithm, which determines the DoFs, which need this
        * sign change for a given cell.
        */
+      template <int spacedim>
       void
       get_face_sign_change_rt(const dealii::Triangulation<1>::cell_iterator &,
-                              const unsigned int,
-                              std::vector<double> &face_sign)
+                              const FiniteElement<1, spacedim> &,
+                              const std::vector<MappingType> &,
+                              std::vector<double> &)
       {
         // nothing to do in 1d
-        std::fill(face_sign.begin(), face_sign.end(), 1.0);
       }
 
 
 
+      //      template<int spacedim>
```

Comment:
I think you could do it with this template argument as well.

Reply:
I only hesitated because on line 73 there's a `const unsigned int spacedim = 2;`

Should I still throw that template in there?

Reply:
Just remove that line. I don't think that the function should behave differently based on the space dimension; only the dimension should matter.

Reply:
Okay I updated that. Would I need to update any other files since I'm turning this into a templated function? I know there are .inst files and they relate with templates but I'm unsure how those things work.

Reply:
No, the function is used only in this one file. The compiler can figure this out. It only gets complicated when you declare a template in one (.h) file, define (i.e., implement) it in a .cc file, and want to use it in another .cc file.